clarify writes to dinfc (#417)
riscv/riscv-cheri: src/debug-integration.adoc

** This only needs doing once after resetting the core.
* The <<m_bit>> is used on debug mode entry to determine which CHERI execution mode to enter.

The <<m_bit>> is the only writeable field in <<dinfc>>:

> * It is only updated on a <<CSRRW>> write _if_ the capability being written has a valid <<m_bit>>.

Reviewer: ```suggestion
* It is only updated on a <<CSRRW>> write _if_ the capability being written is an <<infinite-cap>> capability. 
```
Should we add this constraint (which implies valid permissions encoding with M) or do we just ignore all other bits?

Author: we now have a new approach, I'll put a comment below

> * If any other CSR instruction is used, or if the written <<m_bit>> is not valid then <<dinfc>> is not updated.

Reviewer: ```suggestion
* If any other CSR instruction is used, or if the written capability is not <<infinite-cap>> then <<dinfc>> is not updated.
```

Author: we now have a new approach, I'll put a comment below
